### DISCUSSION TITLE
Dovecot will not start in Monterey

### DISCUSSION THREAD

**Discussion by dfelicia** [Homebrew/discussions#2360] [Everyday usage]
After upgrading to Monterey (12.01), Dovecot no longer starts:

```
$ sudo dovecot -F
Oct 27 10:11:18 service(log): Fatal: setrlimit(RLIMIT_DATA, 268435456): Invalid argument
```

I've rebuilt from source to no avail via: `brew install --build-from-source dovecot`

```
$ ulimit -Ha
core file size          (blocks, -c) unlimited
data seg size           (kbytes, -d) unlimited
file size               (blocks, -f) unlimited
max locked memory       (kbytes, -l) unlimited
max memory size         (kbytes, -m) unlimited
open files                      (-n) unlimited
pipe size            (512 bytes, -p) 1
stack size              (kbytes, -s) 65532
cpu time               (seconds, -t) unlimited
max user processes              (-u) 2784
virtual memory          (kbytes, -v) unlimited
```

```
$ brew config
HOMEBREW_VERSION: 3.3.1-31-ged07132
ORIGIN: https://github.com/Homebrew/brew
HEAD: ed07132a3cc94110e8fbb9c412a9c9a56a51bcd4
Last commit: 3 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 91a8afa35f7fc9dd77760a7914493cee00be13af
Core tap last commit: 51 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_CORE_GIT_REMOTE: https://github.com/Homebrew/homebrew-core
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.tjF7O5kqoh/org.macosforge.xquartz:0
HOMEBREW_EDITOR: vi
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.8 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit kabylake
Clang: 13.0.0 build 1300
Git: 2.33.1 => /usr/local/bin/git
Curl: 7.77.0 => /usr/bin/curl
macOS: 12.0.1-x86_64
CLT: N/A
Xcode: 13.1
XQuartz: 2.7.11 => /opt/X11
```

```
$ brew doctor
Your system is ready to brew.
```

```
$ sw_vers
ProductName:	macOS
ProductVersion:	12.0.1
BuildVersion:	21A559
```

**Comment by carlocab** [comment]
Please report this to the Dovecot developers: https://www.dovecot.org/support/. It's very likely that only they know how to fix this.

**Comment by gromgit** [comment]
Since you're running Dovecot as root, what's the output of `sudo ulimit -Ha`?

**Reply by dfelicia** [reply]
```
$ sudo ulimit -Ha
core file size          (blocks, -c) unlimited
data seg size           (kbytes, -d) unlimited
file size               (blocks, -f) unlimited
max locked memory       (kbytes, -l) unlimited
max memory size         (kbytes, -m) unlimited
open files                      (-n) unlimited
pipe size            (512 bytes, -p) 1
stack size              (kbytes, -s) 65532
cpu time               (seconds, -t) unlimited
max user processes              (-u) 2784
virtual memory          (kbytes, -v) unlimited
```

**Comment by dfelicia** [comment]
Workaround is to add this to local.conf:

`default_vsz_limit = 0`

Alerted Dovecot mainters via their mailing list, and they will seek a better solution.

**Reply by timwuhaotian** [reply]
Fatal: Error in configuration file /opt/homebrew/etc/dovecot/dovecot.conf: service(anvil): vsz_limit is too low

**Comment by quaeler** [comment]
For me, the above work around did not work, and I had to excise code as another person [stated they did on that same Dovecot thread.](https://www.mail-archive.com/dovecot@dovecot.org/msg85160.html)

Related to this, what does a green checkmark next to Monterey on the [formula web page](https://formulae.brew.sh/formula/dovecot) really actually mean? That it simply built successfully?

**Reply by SMillerDev** [reply]
That it build and the automated tests ran successfully.

**Reply by quaeler** [reply]
I see; so one can infer from that, that there is neither a unit test touching the code affecting resource limit setting, nor is there an ~integration test requiring the execution of the built binaries, i suppose.

Thanks for the feedback.

**Reply by SMillerDev** [reply]
I'm guessing that the 2k yearly installs include at least more than you two as users of Monterey, and since the tests do execute the binaries I expect there is something different in your system.
